datafusion/common/src/tree_node.rs

```diff
@@ -342,18 +355,21 @@ pub trait DynTreeNode {
 /// Blanket implementation for Arc for any tye that implements
 /// [`DynTreeNode`] (such as [`Arc<dyn PhysicalExpr>`])
 impl<T: DynTreeNode + ?Sized> TreeNode for Arc<T> {
+    fn children_nodes(&self) -> Vec<Arc<T>> {
+        self.arc_children()
+    }
+
     fn apply_children<F>(&self, op: &mut F) -> Result<VisitRecursion>
     where
         F: FnMut(&Self) -> Result<VisitRecursion>,
     {
-        for child in self.arc_children() {
-            match op(&child)? {
+        for child in &self.arc_children() {
```

Comment:
Do we need this 2nd `apply_children()` implementation here? The base implementation in `TreeNode` might be enough.

Reply:
Yea, missed this. This can be removed too.